### PR TITLE
feat: Add tabs for pipeline UI

### PR DESCRIPTION
**Pull request by tiina303**
## Problem

Set up tabs and navigation on the page, so I won't get any merge conflicts working on them separately.
Note: this took way too long, we have so many different ways we do tabs in the app, this seems like the reasonable way to me, so if that's it I'll add a storybook doc for adding tabs to the scene.

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/scenes/pipeline/Pipeline.tsx [hunk 1]
```diff
@@ -1,11 +1,35 @@
...
 
 export function Pipeline(): JSX.Element {
+    const { currentTab } = useValues(pipelineLogic)
+
+    const singular = singularName(currentTab)
     return (
         <div className="pipeline-scene">
-            <PageHeader title="Pipeline" />
+            <PageHeader
+                title="Pipeline"
```
Comment (corywatilo) [inline frontend/src/scenes/pipeline/Pipeline.tsx:18]: Should this be labeled "CDP"?

Comment (tiina303) [reply frontend/src/scenes/pipeline/Pipeline.tsx:18]: no acronyms :)

**Review by xvello — APPROVED** [review]

File: frontend/src/scenes/pipeline/pipelineLogic.tsx [hunk 2]
```diff
@@ -1,6 +1,67 @@
...
-    path: ['scenes', 'pipeline', 'pipelineLogic'],
-})
+export const singularName = (tab: PipelineTabs): string => {
+    switch (tab) {
+        case PipelineTabs.Filtering:
+            return 'filter'
+        case PipelineTabs.Transformations:
+            return 'transformation'
+        case PipelineTabs.Destinations:
+            return 'destination'
```
Comment (tomasfarias) [inline frontend/src/scenes/pipeline/pipelineLogic.tsx:14]: nit: Maybe we should make these all verbs or nouns:
Option 1 (nouns): filters, transformations, destinations
Option 2 (verbs): filter, transform, export?

File: frontend/src/scenes/pipeline/Pipeline.tsx (hunk 1 above)
Comment (corywatilo) [reply frontend/src/scenes/pipeline/Pipeline.tsx:18]: If this is, in fact, the same product as what we have [on the website](https://posthog.com/docs/cdp), we should at least pluralize it so it partially matches: ~Customer Data~ Pipelines

**Review by Twixes — APPROVED** [review]
Seems about right

File: frontend/src/scenes/pipeline/Pipeline.tsx (hunk 1 above)
Comment (tiina303) [reply frontend/src/scenes/pipeline/Pipeline.tsx:18]: CDP = customer data platform (singular). I find pipelines weird as we don't have several atm.

**Pull request merged**